packages/frontend-v2/utils/course/isEqualCourses.ts

```diff
@@ -1,8 +1,14 @@
-import { ScheduleCourse2 } from "@graduate/common";
+/** Only the details we need to compare courses. */
+type CourseEqualsDetails = {
+  classId: string;
+  subject: string;
+} & {
+  [key: string]: any;
+};
```

Comment:
kinda weird, but essentially allows us to pass in any object that has classId and subject. it's nice so that we can use this when we just have classId and subject annnnd we don't have to extract classId and subject from ScheduleCourse if we have the actual schedule course